compiler/kernel/proc.go

```diff
@@ -590,47 +586,22 @@ func (b *Builder) PushdownOf(trunk *dag.Trunk) (*Filter, error) {
 	return filter, nil
 }
 
-func (b *Builder) LoadConsts(ops []dag.Op) error {
-	scope := b.scope
-	zctx := b.pctx.Zctx
-	for _, p := range ops {
-		switch p := p.(type) {
-		case *dag.Const:
-			val, err := evalAtCompileTime(zctx, scope, p.Expr)
-			if err != nil {
-				return err
-			}
-			if val.IsError() {
-				return fmt.Errorf("cannot resolve const '%s' at compile time", p.Name)
-			}
-			scope.Bind(p.Name, val)
-		case *dag.TypeProc:
-			name := p.Name
-			typ, err := zson.TranslateType(zctx, p.Type)
-			if err != nil {
-				return err
-			}
-			alias, err := zctx.LookupTypeAlias(name, typ)
-			if err != nil {
-				return err
-			}
-			zv := zed.NewTypeValue(alias)
-			scope.Bind(name, zv)
-		default:
-			return fmt.Errorf("kernel.LoadConsts: not a const: '%T'", p)
-		}
-	}
-	return nil
-}
-
-func evalAtCompileTime(zctx *zed.Context, scope *Scope, in dag.Expr) (*zed.Value, error) {
+func EvalAtCompileTime(zctx *zed.Context, in dag.Expr) (val *zed.Value, err error) {
 	if in == nil {
 		return zed.Null, nil
 	}
-	e, err := compileExpr(zctx, scope, in)
+	e, err := compileExpr(zctx, in)
 	if err != nil {
 		return nil, err
 	}
+	// Catch panic as the runtime will panic if there is a
+	// reference to a var not in scope, a field access null this, etc
+	defer func() {
+		if recover() != nil {
+			val = nil
+			err = errors.New("panic")
```

Comment:
```suggestion
		if e := recover(); e != nil {
			err = fmt.Errorf("%+v", e)
```

Reply:
This error value is always ignored and the error conditions is used to detect an non-constant expression (i.e., not dependent on flowgraph/this).    Thus, it expects to typically panic on this == nil as all the expr code assumes this != nil.  The right way to make this better would be to carefully make sure there is a detectable difference between unrecoverable errors encountered here vs expected panics due to null this and for unrecoverable errors report them as suggested here.  This seems low priority.